### DISCUSSION TITLE
Kit errors out on running scripts

### DISCUSSION THREAD

**Discussion by lsminter** [johnlindquist/kit#387] [Errors]
This is the error I'm receiving every time I try to run the dad joke script. 
```
Request failed with status code 503 Error: Request failed with status code 503 at createError 
(/Users/lucasminter/.kit/node_modules/axios/lib/core/createError.js:16:15) at settle 
(/Users/lucasminter/.kit/node_modules/axios/lib/core/settle.js:17:12) at 
IncomingMessage.handleStreamEnd 
(/Users/lucasminter/.kit/node_modules/axios/lib/adapters/http.js:260:11) at IncomingMessage.emit 
(node:events:406:35) at endReadableNT (node:internal/streams/readable:1331:12) at 
processTicksAndRejections (node:internal/process/task_queues:83:21)
```

It seems like when I try to get access to an external dependency, I get an error. 

I've tried deleting my .kenv and my .kit and reinstalling from scratch and no luck. 

I've also had issues trying to open it in an editor. I would get a quick error, too fast to see, then kit would close. After deleting everything and reinstalling, I don't have an option to open in VSCode, only Vim and Nano.

**Comment by lsminter** [comment]
I've got it opening in the correct editor now. I'm unsure how code showed up now when selecting a way to edit my code. It just sorta worked after retrying 20 minutes later. Still having the issue with axios though.

**Reply by johnlindquist** [reply]
It's some `axios` issue I'm trying to track down. My guess is a peer dependency updated or something.

In a vanilla node.js project, this gets the same 503 error:
```js
var axios = require("axios")

var config = {
  method: "get",
  url: "https://icanhazdadjoke.com/",
  headers: {
    Accept: "text/plain",
  },
}

axios(config)
  .then(function (response) {
    console.log(JSON.stringify(response.data))
  })
  .catch(function (error) {
    console.log(error)
  })
```

**Reply by johnlindquist** [reply]
Looks like the server requires a "User-Agent" now. This should work:

```js
// Menu: Dad Joke
// Description: Dad Joke from icanhazdadjoke.com
// Author: John Lindquist
// Twitter: @johnlindquist

let response = await get(`https://icanhazdadjoke.com/`, {
  headers: {
    Accept: "text/plain",
    "User-Agent": "axios 0.21.1",
  },
})

await div(response.data, `p-4`)
```

**Reply by lsminter** [reply]
Yep! Works for me!

**Comment by johnlindquist** [comment]
Kit searches for installed editors using `which` like so:

```bash
PATH="/usr/bin:/usr/local/bin" which code
```

I'll try to think if there's a more reliable way. I'm not sure how this would have failed.

Worst case scenario is you manually open:

`~.kenv/.env`

and add:

```bash
KIT_EDITOR=code
```

**Reply by lsminter** [reply]
Yeah I'm not sure how it got fixed. I wish I had an answer for you.
